### PR TITLE
[ADD] Cart and purchase

### PR DESCRIPTION
**Pull request by DanielHernandezO**
Added cart logic and fixed lag, migrations and model's relations

### CODE REVIEW

File: app/Http/Controllers/UserCartController.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+use App\Models\Order;
+use App\Models\OrderItem;
+use Illuminate\Http\Request;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\View\View;
+use Illuminate\Support\Facades\Auth;
+
+class UserCartController extends Controller
+{
+    public function index(Request $request): view
```
Comment (Jguerra47) [inline app/Http/Controllers/UserCartController.php:15]: Return type must start with capital case

File: app/Models/Item.php [hunk 2]
```diff
@@ -39,6 +39,15 @@ public static function validate(Request $request): void
         ]);
     }
 
+    public static function sumPricesByQuantities($items, $itemsInSession): int
```
Comment (Jguerra47) [inline app/Models/Item.php:42]: Attributes type missing

File: app/Models/Order.php [hunk 3]
```diff
@@ -55,6 +64,16 @@ public function getUpdatedAt(): string
         return $this->attributes['updated_at'];
     }
 
+    public function getUserid(): int
```
Comment (Jguerra47) [inline app/Models/Order.php:67]:
```suggestion
    public function getUserId(): int
```

**Review by Jguerra47 — COMMENTED** [review]
Some typo comments

File: app/Http/Controllers/UserCartController.php (hunk 1 above)
Comment (DanielHernandezO) [reply app/Http/Controllers/UserCartController.php]: Done

**Review by Jguerra47 — APPROVED** [review]

**Pull request merged**